core/src/main/java/bisq/core/offer/takeoffer/TakeOfferModel.java

```diff
@@ -0,0 +1,293 @@
+package bisq.core.offer.takeoffer;
+
+import bisq.core.account.witness.AccountAgeWitnessService;
+import bisq.core.btc.model.AddressEntry;
+import bisq.core.btc.wallet.BtcWalletService;
+import bisq.core.locale.CurrencyUtil;
+import bisq.core.monetary.Price;
+import bisq.core.monetary.Volume;
+import bisq.core.offer.Offer;
+import bisq.core.offer.OfferUtil;
+import bisq.core.payment.PaymentAccount;
+import bisq.core.payment.payload.PaymentMethod;
+import bisq.core.provider.fee.FeeService;
+import bisq.core.provider.price.PriceFeedService;
+
+import bisq.common.taskrunner.Model;
+
+import org.bitcoinj.core.Coin;
+
+import javax.inject.Inject;
+
+import java.util.Objects;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.jetbrains.annotations.NotNull;
+
+import static bisq.core.btc.model.AddressEntry.Context.OFFER_FUNDING;
+import static bisq.core.offer.OfferPayload.Direction.SELL;
+import static bisq.core.util.VolumeUtil.getAdjustedVolumeForHalCash;
+import static bisq.core.util.VolumeUtil.getRoundedFiatVolume;
+import static bisq.core.util.coin.CoinUtil.minCoin;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.bitcoinj.core.Coin.ZERO;
+import static org.bitcoinj.core.Coin.valueOf;
+
+@Slf4j
+public class TakeOfferModel implements Model {
+    // Immutable
+    private final AccountAgeWitnessService accountAgeWitnessService;
+    private final BtcWalletService btcWalletService;
+    private final FeeService feeService;
+    private final OfferUtil offerUtil;
+    private final PriceFeedService priceFeedService;
+
+    // Mutable
+    @Getter
+    private AddressEntry addressEntry;
+    @Getter
+    private Coin amount;
+    @Getter
+    private boolean isCurrencyForTakerFeeBtc;
+    private Offer offer;
+    private PaymentAccount paymentAccount;
+    @Getter
+    private Coin securityDeposit;
+    private boolean useSavingsWallet;
+
+    // 260 kb is typical trade fee tx size with 1 input, but trade tx (deposit + payout)
+    // are larger so we adjust to 320.
+    private final int feeTxSize = 320;
+    private Coin txFeePerByteFromFeeService;
+    @Getter
+    private Coin txFeeFromFeeService;
+    @Getter
+    private Coin takerFee;
+    @Getter
+    private Coin totalToPayAsCoin;
+    @Getter
+    private Coin missingCoin = ZERO;
+    @Getter
+    private Coin totalAvailableBalance;
+    @Getter
+    private Coin balance;
+    @Getter
+    private boolean isBtcWalletFunded;
+    @Getter
+    private Volume volume;
+
+    @Inject
+    public TakeOfferModel(AccountAgeWitnessService accountAgeWitnessService,
```

Comment:
There's a lot of code that would be nice to reuse from `TakeOfferDataModel` here. Can be done later though in some pure refactor commit.

Reply:
I mentioned that I wanted to move more from `TakeOfferDataModel into core `TakeOfferModel`, but I could not pass JFX property values into the new core model.   If you or @chimp1984 know how, do tell.

Reply:
Don't think we want to have any jfx dependency in core, rather some stuff could probably be rewritten in `TakeOfferDataModel` to utilize the new class.